docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,324 @@
+# Developer guide for Usage of MPU Regions in TizenRT
+
+## CONTENTS
+- [Purpose of MPU](#purpose-of-mpu)
+- [ARM MPU Attributes and Configuration](#arm-mpu-attributes-and-configuration)
+- [Prerequisites for MPU Usage in TizenRT](#prerequisites-for-mpu-usage-in-tizenrt)
+- [Functions used and Enumeration for MPU regions in TizenRT](#functions-used-and-enumeration-for-mpu-regions-in-tizenrt)
+- [How to Control MPU regions in TizenRT](#how-to-control-mpu-regions-in-tizenrt)
+- [Current Usage of MPU Regions in TizenRT](#current-usage-of-mpu-regions-in-tizenrt)
+- [Generic memory layout of existing MPU regions](#generic-memory-layout-of-existing-mpu-regions)
+
+## Purpose of MPU
+
+- Please refer to below ARM website link for understanding purpose of MPU:
+	- [About the MPU](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.ddi0439b/Chdecfea.html)
+
+## ARM MPU Attributes and Configuration
+
+- Currently, only ARMV7M and ARMV8M MPUs are supported in TizenRT.
+- ARMV8M MPU has following features:
+	- The MPU supports a configurable number of programmable regions between zero and eight regions per security state.
+	- The smallest size that can be programmed for an MPU region is 32 bytes.
+	- The maximum size of any MPU region is 4GB, but the size must be a multiple of 32 bytes.
+	- All regions must begin on a 32 byte aligned address.
+
```

Comment:
Please mention the features of armv7m mpu also. 
It is important because there are restrictions of region size being power of 2 and region start address being aligned according to size. Many of our design considerations are based on these restrictions.

Reply:
I dont think support for ARMv8M MPU support is verified even without security extension. 
Also, with security extension, it requires modelling with SAU and IDAU for memory partition created. And will be banked across the security states. 
I suggest, it may not be correct to highlight ARMv8M at the current state here.

Reply:
Done.
Mentioned features for ARMv7M.
Also, mentioned:
`Currently, ARMv8M support is not verified in TizenRT.`